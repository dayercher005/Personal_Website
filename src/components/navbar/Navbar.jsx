import Title from './Title.jsx'

export default function Navbar() {

    const portfolioName = "Dayer Cher"

    return (
        <nav className="flex justify-evenly p-5 sticky top-0 backdrop-blur-sm">

            <h1 className="basis-64 font-bold text-3xl">{portfolioName}</h1>

            <div className="flex basis-128 justify-evenly rounded-2xl gap-4">
                <Title name="Home"></Title>
                <Title name="About"></Title>
                <Title name="Projects"></Title>
                <Title name="Skills"></Title>
                <Title name="Contact"></Title>
            </div>
        </nav>
    )
}