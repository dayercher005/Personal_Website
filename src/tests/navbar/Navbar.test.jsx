import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Navbar from '/src/components/navbar/Navbar.jsx'

describe("Navbar component", () => {
    it("renders correct heading of NavBar", () => {
        render(<Navbar/>)
        expect(screen.getByRole("heading", {name: "Dayer Cher"}))
    })

    it("renders entire Navbar component and correct titles of Navbar", () => {
        const { container } = render(<Navbar/>)
        expect(container).toMatchSnapshot()
    })
})