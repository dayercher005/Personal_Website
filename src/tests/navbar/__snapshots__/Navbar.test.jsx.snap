// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > renders correct titles of Navbar 1`] = `
<div>
  <nav
    class="flex justify-evenly p-5 sticky top-0 backdrop-blur-sm"
  >
    <h1
      class="basis-64 font-bold text-3xl"
    >
      Dayer Cher
    </h1>
    <div
      class="flex basis-128 justify-evenly rounded-2xl gap-4"
    >
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Home"
      >
        Home
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#About"
      >
        About
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Projects"
      >
        Projects
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Skills"
      >
        Skills
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Contact"
      >
        Contact
      </a>
    </div>
  </nav>
</div>
`;

exports[`Navbar component > renders entire Navbar component and correct titles of Navbar 1`] = `
<div>
  <nav
    class="flex justify-evenly p-5 sticky top-0 backdrop-blur-sm"
  >
    <h1
      class="basis-64 font-bold text-3xl"
    >
      Dayer Cher
    </h1>
    <div
      class="flex basis-128 justify-evenly rounded-2xl gap-4"
    >
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Home"
      >
        Home
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#About"
      >
        About
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Projects"
      >
        Projects
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Skills"
      >
        Skills
      </a>
      <a
        class="hover:scale-105 flex items-center text-xl font-medium px-5 rounded-2xl hover:bg-gray-300 transition transition-300"
        href="#Contact"
      >
        Contact
      </a>
    </div>
  </nav>
</div>
`;
